src/app/shared/components/label-form/component.ts

```diff
@@ -96,18 +103,44 @@ export class LabelFormComponent implements OnChanges, OnInit, OnDestroy, Control
   }
 
   ngOnChanges(changes: SimpleChanges) {
+    this.labels = this.labels ?? {};
+
+    this.inheritedLabels = this.inheritedLabels ?? {};
+
+    if (!this.form) {
+      this._initForm();
+    }
     if (changes && changes.keyName) {
       this.keyName = changes.keyName.currentValue;
     }
     if (changes && changes.valueName) {
       this.valueName = changes.valueName.currentValue;
     }
+    if (changes?.disabledLabel) {
+      const [key = '', value = ''] = this.disabledLabel ?? [];
+
+      if (this.disabledLabelFormGroup) {
+        this.disabledLabelFormGroup.patchValue({key, value});
+      } else {
+        this.disabledLabelFormGroup = this._formBuilder.group({
```

Comment:
I think if `key` and `value` are not defined (empty or undefined) then we should set this form group to `null`. Wdyt?

Reply:
the values of these form groups are being set to `undefined`, this whole form group is not being set to `null`.

Reply:
Yes. but should we set this whole form group to `null`?

Reply:
I don't think that's necessary.